matdeeplearn/trainers/property_trainer.py

```diff
@@ -37,22 +40,32 @@ def __init__(
             test_loader,
             loss,
             max_epochs,
+            max_checkpoint_epochs,
             identifier,
             verbosity,
+            save_dir,
+            checkpoint_dir,
         )
 
     def train(self):
+        # Start training over epochs loop
+        # Calculate start_epoch from step instead of loading the epoch number
+        # to prevent inconsistencies due to different batch size in checkpoint.
+        start_epoch = self.step // len(self.train_loader)
+
+        epochs_to_run = (
+            self.max_checkpoint_epochs + start_epoch
+            if self.max_checkpoint_epochs
+            else self.max_epochs
+        )
+
         if self.train_verbosity:
             logging.info("Starting regular training")
             logging.info(
-                f"running for  {self.max_epochs} epochs on {type(self.model).__name__} model"
+                f"running for {epochs_to_run} epochs on {type(self.model).__name__} model"
```

Comment:
If you make the change I suggested in the comment below, then this should be
```suggestion
                f"running for {end_epoch - start_epoch} epochs on {type(self.model).__name__} model"
```